Sources/Host/Common/Socket.cpp

```diff
@@ -99,11 +99,12 @@ bool Socket::listen(std::string const &address, std::string const &port) {
   if (listening() || connected())
     return false;
 
+  // Disable socket lingering so the process can die quickly when we exit.
   struct linger linger;
-  linger.l_onoff = 1;
+  linger.l_onoff = 0;
   linger.l_linger = 0;
   ::setsockopt(_handle, SOL_SOCKET, SO_LINGER,
-               reinterpret_cast<char *>(&linger), sizeof(linger));
+               reinterpret_cast<void *>(&linger), sizeof(linger));
```

Comment:
You can pass a pointer as argument to a function that takes a const pointer. The constness of the function argument just indicates that the callee will not modify the data; the pointer you hold being const makes no difference.